presto-testing/src/main/java/io/prestosql/testing/AbstractTestJoinQueries.java

```diff
@@ -2275,6 +2275,39 @@ public void testEquijoinOnDifferentTypesWithFilter()
                 "WHERE id3 = 10");
     }
 
+    @Test
+    public void testMultiJoinWithEligibleForDynamicFiltering()
+    {
+        assertQuery("" +
+                        "SELECT customer.name, lineitem.partkey " +
+                        "FROM lineitem " +
+                        "LEFT JOIN orders ON lineitem.orderkey = orders.orderkey " +
```

Comment:
BTW 
With DF, we are able to convert this join to INNER.

Without DF we could be able too -- @sopel39 do you know where this could be happening (but isn't)?

If we can pull this off, this would fix these test queries, but I would still think we should have the safety check in PPD, the one I'm adding in this PR.

Reply:
This is probably due to limitations is `PredicatePushdown.processLimitedOuterJoin()`. I made some improvements to `processInnerJoin()` a while ago, but they are probably missing from that other method.

Reply:
> Without DF we could be able too -- @sopel39 do you know where this could be happening (but isn't)?

dynamic filter is predicate on symbol `dynamic_filter(symbol, "dyn_id")`. when `symbol` is null, then predicate result is also `null`. Therefore PPD is able to convert some of outer joins to inner joins. This is similar to https://github.com/prestosql/presto/pull/2415, but https://github.com/prestosql/presto/pull/2415 has problems as additional `NOT NULL` check can be expensive sometimes (when not needed)

Reply:
@sopel39 exactly

@martint i will look into `processLimitedOuterJoin`, but i think we should still have this safety check

Reply:
The simplest query where LEFT could be converted into INNER that i could come up with is:
```
SELECT customer.name, lineitem.partkey
FROM lineitem
LEFT JOIN orders ON lineitem.orderkey = orders.orderkey
INNER JOIN customer ON orders.custkey = customer.custkey
WHERE customer.name IS NOT NULL
```

Indeed, #2415 is necessary (and sufficient) for this.

Without that PR, during PPD, the LEFT join in this query doesn't get any effective predicate that could be used to convert it into an INNER.
Which brings us back to square one.